tools/rest_api.rb

```diff
@@ -1,293 +1,6 @@
-
-          msg_exit("Action #{action} requires data to be specified") if data.empty?
-        end
-      end
-
-      conn = Faraday.new(:url => opts[:url], :ssl => {:verify => false}) do |faraday|
-        faraday.request(:url_encoded)               # form-encode POST params
-        faraday.response(:logger) if opts[:verbose] # log requests to STDOUT
-        faraday.use FaradayMiddleware::FollowRedirects, :limit => 3, :standards_compliant => true
-        faraday.adapter(Faraday.default_adapter)    # make requests with Net::HTTP
-        faraday.basic_auth(opts[:user], opts[:password]) if opts[:token].empty? && opts[:miqtoken].empty?
-      end
-
-      if action == "run"
-        puts "Loading #{api_script}"
-        require api_script
-        as = ApiScript.new(CTYPE, conn)
-        puts "Running #{api_script} with method #{method} ..."
-        method.nil? ? as.run : as.run(method)
-        exit
-      end
-
-      path = PREFIX.dup
-      path << "/v#{opts[:apiversion]}" unless opts[:apiversion].empty?
-
-      collection = ""
-      item = ""
-      unless resource.nil?
-        path << resource
-        rscan = resource.scan(%r{[^/]+})
-        collection, item = rscan[0..1]
-      end
-
-      if opts[:verbose]
-        puts SEP
-        puts "Connection Endpoint: #{opts[:url]}"
-        puts "Action:              #{action}"
-        puts "HTTP Method:         #{method}"
-        puts "Resource:            #{resource}"
-        puts "Collection:          #{collection}"
-        puts "Item:                #{item}"
-        puts "Parameters:"
-        params.keys.each { |k| puts "#{' ' * 21}#{k} = #{params[k]}" }
-        puts "Path:                #{path}"
-        puts "Data:                #{data}"
-      end
-
-      begin
-        response = conn.run_request(method.to_sym, path, nil, nil) do |req|
-          req.headers[:content_type]  = CTYPE
-          req.headers[:accept]        = CTYPE
-          req.headers['X-MIQ-Group']  = opts[:group] unless opts[:group].empty?
-          req.headers['X-MIQ-Token']  = opts[:miqtoken] unless opts[:miqtoken].empty?
-          req.headers['X-Auth-Token'] = opts[:token] unless opts[:token].empty?
-          req.params.merge!(params)
-          req.body = data if METHODS_NEEDING_DATA.include?(method)
-        end
-      rescue => e
-        msg_exit("\nFailed to connect to #{opts[:url]} - #{e}")
-      end
-
-      if opts[:verbose]
-        puts SEP
-        puts "Response Headers:"
-        puts response.headers
-
-        puts SEP
-        puts "Response Body:"
-      end
-
-      if response.body
-        body = response.body.strip
-        if opts[:format] == "pretty"
-          puts json_pretty(body) unless body.empty?
-        else
-          puts body
-        end
-      end
-
-      exit response.status >= 400 ? 1 : 0
-    end
-  end
-end
-RestApi::Cli.new.run
+gem_dir = Pathname.new(Bundler.locked_gems.specs.select { |g| g.name == "manageiq-api" }.first.gem_dir)
```

Comment:
We keep losing your comment @NickLaMuro 

> Interested to see how it works out. The previous implementation is definitely a more of a public API than what I suggested, but this might make sure we are going through bundler.
> Anyway, was mostly a suggestion, and if it causes problems, feel free to use the old code.

We're already going through `bundler/setup` on line 2 so it's safe to assume we need to run through bundler.  @abellotti is that a safe assumption?  That they are going to use the gems in the rails app bundler?